internal/pkg/config/root.go

```diff
@@ -99,6 +99,15 @@ func (conf *RootConf) Parse(data []byte) error {
 	if len(conf.TFTP.IpxeBinaries) == 0 {
 		conf.TFTP.IpxeBinaries = defIpxe
 	}
+
+	// check whether ip addr is CIDR type and configure related fields as required
+	if ip, network, err := net.ParseCIDR(conf.Ipaddr); err == nil {
+		conf.Ipaddr = ip.String()
+		conf.Network = network.IP.String()
+		mask := network.Mask
+		conf.Netmask = fmt.Sprintf("%d.%d.%d.%d", mask[0], mask[1], mask[2], mask[3])
```

Comment:
conf.Network and conf.Netmask should only be populated here if they don't have their own value specified in the config file.